internal/internal_workflow.go

```diff
@@ -444,32 +456,42 @@ func newWorkflowContext(env WorkflowEnvironment, interceptors WorkflowIntercepto
 	return rootCtx
 }
 
-func newWorkflowInterceptors(env WorkflowEnvironment, factories []WorkflowInterceptorFactory) (WorkflowInterceptor, *workflowEnvironmentInterceptor) {
+func newWorkflowInterceptors(env WorkflowEnvironment, interceptors []WorkflowInterceptor) (*workflowEnvironmentInterceptor, error) {
```

Comment:
I think the factory pattern was better in this case as we are establishing an interceptor chain based on factories registered with the registry.

Reply:
Factory is confusing IMHO as we are intercepting workflow executions